### PR TITLE
Return allowed actions with visualization responses

### PR DESCRIPTION
**Pull request by cameronmarsh**

### CODE REVIEW

File: apps/discovery_api/lib/discovery_api_web/controllers/visualization_controller.ex [hunk 1]
```diff
@@ -5,20 +5,23 @@ defmodule DiscoveryApiWeb.VisualizationController do
...
          visualizations <- Visualizations.get_visualizations_by_owner_id(user.id) do
-      render(conn, :visualizations, %{visualizations: visualizations})
+      render(conn, :visualizations, %{visualizations: visualizations, allowed_actions: @owner_allowed_actions})
     end
   end
 
   def show(conn, %{"id" => id}) do
     with {:ok, %{query: query} = visualization} <- Visualizations.get_visualization_by_id(id),
-         user <- Map.get(conn.assigns, :current_user),
+         user = Map.get(conn.assigns, :current_user),
```
Comment (kcarra) [inline apps/discovery_api/lib/discovery_api_web/controllers/visualization_controller.ex]: Perfect usage of this - Ben

File: apps/discovery_api/lib/discovery_api_web/controllers/visualization_controller.ex [hunk 2]
```diff
@@ -59,4 +65,13 @@ defmodule DiscoveryApiWeb.VisualizationController do
   defp under_visualizations_limit?(visualizations) do
     Enum.count(visualizations) <= Application.get_env(:discovery_api, :user_visualization_limit)
   end
+
+  defp get_allowed_actions(_visualization, nil), do: []
```
Comment (bbrewer-papa-x) [inline apps/discovery_api/lib/discovery_api_web/controllers/visualization_controller.ex]: Is there any more appropriate layer for this kind of thing than here? I think it's probably fine, and I'm not sure if it really belongs in the BL layer either.

File: apps/discovery_api/lib/discovery_api_web/controllers/visualization_controller.ex (hunk 1 above)
Comment (bbrewer-papa-x) [inline apps/discovery_api/lib/discovery_api_web/controllers/visualization_controller.ex]: Perfect usage of this

File: apps/discovery_api/test/unit/discovery_api_web/controllers/visualization_controller_test.exs [hunk 3]
```diff
@@ -121,7 +114,8 @@ defmodule DiscoveryApiWeb.VisualizationControllerTest do
                "query" => @query,
                "title" => @title,
                "id" => @id,
-               "chart" => @decoded_chart
+               "chart" => @decoded_chart,
+               "allowed_actions" => [%{"name" => "update"}, %{"name" => "create_copy"}]
```
Comment (bbrewer-papa-x) [inline apps/discovery_api/test/unit/discovery_api_web/controllers/visualization_controller_test.exs]: Thank you for choosing this form for the return. We've chosen things like arrays of strings and maps before and it has locked us in pretty hard.

File: apps/discovery_api/lib/discovery_api_web/controllers/visualization_controller.ex (hunk 2 above)
Comment (mcsearchin) [reply apps/discovery_api/lib/discovery_api_web/controllers/visualization_controller.ex]: Good call out.  I'm thinking we wait and see how much more complicated this gets.  We could move it into its own plug or a helper module, but probably wouldn't do so until we have more rules.

File: apps/discovery_api/test/unit/discovery_api_web/controllers/visualization_controller_test.exs (hunk 3 above)
Comment (mcsearchin) [reply apps/discovery_api/test/unit/discovery_api_web/controllers/visualization_controller_test.exs]: 👍 most of the credit to @JarredOlson

**Pull request merged**